libgalois/include/katana/PropertyGraph.h

```diff
@@ -694,6 +710,33 @@ class KATANA_EXPORT PropertyGraph {
     return edge_entity_type_manager_;
   }
 
+  /// Get the version from the RDGFile
+  katana::RDGVersion RDGFileVersion() {
+    if (file_ == nullptr) {
+      return katana::RDGVersion(0);
+    } else {
+      return rdg_.GetFileVersion(*file_);
+    }
+  }
+
+  /// Set the targeted branch in graph_dir for writing PropertyGraph
+  /// If a positive num is provided, new graph should retain the version number
+  void SetBranch(const katana::RDGVersion& val, uint64_t num = 0) {
+    branch_ = val;
+    branch_.SetLeafNumber(num);
+  }
+  /// Get the targeted branch in graph_dir for writing PropertyGraph
+  katana::RDGVersion& GetBranch() { return branch_; }
+
+  /// Record the version of loaded PropertyGraph
+  void SetLoadedVersion(const katana::RDGVersion& val) {
+    loaded_version_ = val;
+  }
+  /// Get the version of loaded PropertyGraph (const)
+  const katana::RDGVersion& GetLoadedVersion() const { return loaded_version_; }
+  /// Get the version of loaded PropertyGraph for using its member functions
+  katana::RDGVersion& GetLoadedVersion() { return loaded_version_; }
```

Comment:
Having a mutable version just seems really weird and dangerous, why not just require that a user get the value change it and then set it?

Also I don't see why the LoadedVersion would ever change after loading, so this methods seem like they should be private if possible.

Reply:
Not sure I guess correctly. You seem to interpret a version identifier equivalent as an RDG. My conversations with others concluded that a null version is always needed to begin a new RDG. It was previous 0, now updated to RDGVersion(0). As an identifier, it only means that no specific version is targeted for search. 

My next step is to focus on updating the RFC. Not yet have a clear solution on branch_, loaded_version_, etc. It is a bit complicated.

Reply:
So "Null" for RDGVersion is the first version for every graph. That is a reasonable thing to have. I would *much* prefer we call it `Zero`. Since null has special meaning in C++.